src/main/java/tools/jackson/core/io/IOContext.java

```diff
@@ -362,4 +365,11 @@ private IllegalArgumentException wrongBuf() {
         // sanity check failed; trying to return different, smaller buffer.
         return new IllegalArgumentException("Trying to release buffer smaller than original");
     }
+
+    @Override
+    public void close() {
+        if (closed.compareAndSet(false, true)) {
```

Comment:
This is supposed to be concurrent? If not, and just "ordered" you can use a plain bool

Reply:
For now I implemented it in a concurrent way, but I agree that this constraint can be relaxed if not necessary.